server/buildbuddy_server/buildbuddy_server_test.go

```diff
@@ -0,0 +1,202 @@
+package buildbuddy_server_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/buildbuddy-io/buildbuddy/server/build_event_protocol/build_event_handler"
+	"github.com/buildbuddy-io/buildbuddy/server/buildbuddy_server"
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testauth"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/buildbuddy-io/buildbuddy/proto/acl"
+	"github.com/buildbuddy-io/buildbuddy/proto/build_event_stream"
+	bepb "github.com/buildbuddy-io/buildbuddy/proto/build_events"
+	inpb "github.com/buildbuddy-io/buildbuddy/proto/invocation"
+	pepb "github.com/buildbuddy-io/buildbuddy/proto/publish_build_event"
+	"github.com/buildbuddy-io/buildbuddy/proto/user_id"
+)
+
+const (
+	user1  = "USER1"
+	group1 = "GROUP1"
+	user2  = "USER2"
+	group2 = "GROUP2"
+)
+
+func createInvocationForTesting(te environment.Env, user string) (string, error) {
+	ctx := context.Background()
+	testUUID, err := uuid.NewRandom()
+	if err != nil {
+		return "", err
+	}
+	testInvocationID := testUUID.String()
+
+	handler := build_event_handler.NewBuildEventHandler(te)
+	channel := handler.OpenChannel(ctx, testInvocationID)
+
+	// Send started event with api key
+	started, err := anypb.New(&build_event_stream.BuildEvent{
+		Payload: &build_event_stream.BuildEvent_Started{
+			Started: &build_event_stream.BuildStarted{
+				OptionsDescription: "--remote_header='" + testauth.APIKeyHeader + "=" + user + "'",
+			},
+		},
+	})
+	request := &pepb.PublishBuildToolEventStreamRequest{
+		OrderedBuildEvent: &pepb.OrderedBuildEvent{
+			SequenceNumber: 1,
+			StreamId:       &bepb.StreamId{InvocationId: testInvocationID},
+			Event: &bepb.BuildEvent{
+				Event: &bepb.BuildEvent_BazelEvent{
+					BazelEvent: started,
+				},
+			},
+		},
+	}
+	err = channel.HandleEvent(request)
+	if err != nil {
+		return "", err
+	}
+	return testInvocationID, err
+}
+
+func TestGetInvocation(t *testing.T) {
+	te := testenv.GetTestEnv(t)
+	auth := testauth.NewTestAuthenticator(testauth.TestUsers(user1, group1, user2, group2))
+	te.SetAuthenticator(auth)
+
+	iid, err := createInvocationForTesting(te, user1)
+	assert.NoError(t, err)
```

Comment:
You should use `require.NoError` and other `require` functions over their `assert` counterparts when it's not useful to continue the test after an assertion failure. The `assert` functions don't stop the test execution so the test can report multiple errors or terminate incorrectly because of precondition failures.

Reply:
Awesome; thanks for the information as always, Vadim!